modin/backends/base/doc_utils.py

```diff
@@ -0,0 +1,685 @@
+
+    Parameters
+    ----------
+    action : str
+        What method does with the resampled data.
+    output : str
+        What is the content of column names in the result.
+    refer_to : str
+        Method name in ``modin.pandas.base.Resampler`` module to refer to for
+        more information about parameters and output format.
+    params : str, optional
+        Method parameters in the NumPy docstyle format to substitute
+        to the docstring template.
+
+    Returns
+    -------
+    callable
+    """
+    action = f"{action} for each group over the specified axis"
+
+    params_substitution = """
+        *args : iterable
+            Positional arguments to pass to the aggregation function.
+        **kwargs : dict
+            Keyword arguments to pass to the aggregation function.
+        """
+
+    if params:
+        params_substitution = format_string(
+            "{params}\n{params_substitution}",
+            params=params,
+            params_substitution=params_substitution,
+        )
+
+    build_rules = f"""
+            - Labels on the specified axis are the group names (time-stamps)
+            - Labels on the opposit of specified axis are a MultiIndex, where first level
+              contains preserved labels of this axis and the second level is the {output}.
+            - Each element of QueryCompiler is the result of corresponding function for the
+              corresponding group and column/row."""
+    return doc_resample(
+        action=action,
+        extra_params=params_substitution,
+        build_rules=build_rules,
+        refer_to=refer_to,
+    )
+
+
+def doc_resample_fillna(method, refer_to, params=None, overwrite_template_params=False):
+    """
+    Build decorator which adds docstring for the resample fillna query compiler method.
+
+    Parameters
+    ----------
+    method : str
+        Fillna method name.
+    refer_to : str
+        Method name in ``modin.pandas.base.Resampler`` module to refer to for
+        more information about parameters and output format.
+    params : str, optional
+        Method parameters in the NumPy docstyle format to substitute
+        to the docstring template.
+    overwrite_template_params : bool, default: False
+        If `params` is specified indicates whether to overwrite method parameters in
+        the docstring template or append then at the end.
+
+    Returns
+    -------
+    callable
+    """
+    action = f"fill missing values in each group independently using {method} method"
+    params_substitution = "limit : int\n"
+
+    if params:
+        params_substitution = (
+            params
+            if overwrite_template_params
+            else format_string(
+                "{params}\n{params_substitution}",
+                params=params,
+                params_substitution=params_substitution,
+            )
+        )
+
+    build_rules = "- QueryCompiler contains unsampled data with missing values filled."
```

Comment:
could you clarify please what this means?

Reply:
`build_rules` is a string formally describing how the function's output looks like. This one is substitution at the general [resample functions template](https://github.com/dchigarev/modin/blob/016e44f8d54354bdfc75ef8812732807b60ca51e/modin/backends/base/doc_utils.py#L322).